src/plugins/muc-views/utils.js

```diff
@@ -127,9 +127,24 @@ export function getAutoCompleteListItem (text, input) {
     return element;
 }
 
-export async function getAutoCompleteList () {
-    const models = [...(await api.rooms.get()), ...(await api.contacts.get())];
-    const jids = [...new Set(models.map(o => Strophe.getDomainFromJid(o.get('jid'))))];
+// models array is globalized to introduce caching of the list from https://search.jabber.network/rooms/1.
+const models = [];
+let callFetchListOfRooms = false;
+
+async function fetchListOfRooms() {
+    const response = await fetch('https://search.jabber.network/api/1.0/rooms');
+    const data = await response.json();
+    for (var x = 0; x < data.items.length; x++) {
+        models.push(data.items[x]["address"]);
+    }
+}
+
+export function getAutoCompleteList() {
+    if (callFetchListOfRooms === false) {
+        fetchListOfRooms();
+        callFetchListOfRooms = true;
```

Comment:
Instead of setting a boolean, you can set a timestamp and then use DayJS to check whether the current time minus the timestamp is above a certain threshold, (e.g. 4 hours).

Reply:
I would be ok with it, but is it required? the above implementation simply fetches once per session, and it's simple to understand. Do we still need a timestamp and need to fetch the list in every interval? I feel that it is unnecessary, what's your opinion @jcbrand?

Reply:
Using a timestamp is trivial as well. 

Instead of `if (callFetchListOfRooms === false) {` it becomes something like `if (dayjs().isAfter(time_fetched, 'day')) }`

Web sessions are quite short, but as more people start using the Electron version, a session can last many days or weeks.